server/grpc_service.go

```diff
@@ -1990,7 +1991,17 @@ func (s *GrpcServer) WatchGlobalConfig(req *pdpb.WatchGlobalConfigRequest, serve
 			for _, e := range res.Events {
```

Comment:
Perhaps we'd better check the `res.Err()` before handling it? That method can detect both revision compacted and other things made the channel no longer be able to be used.
```go
// Err is the error value if this WatchResponse holds an error.
func (wr *WatchResponse) Err() error {
	switch {
	case wr.closeErr != nil:
		return v3rpc.Error(wr.closeErr)
	case wr.CompactRevision != 0:
		return v3rpc.ErrCompacted
	case wr.Canceled:
		if len(wr.cancelReason) != 0 {
			return v3rpc.Error(status.Error(codes.FailedPrecondition, wr.cancelReason))
		}
		return v3rpc.ErrFutureRev
	}
	return nil
}
```

Reply:
cc @HuSharp

Reply:
done it

Reply:
LGTM